src/Sarif.Driver/Sdk/MultithreadedAnalyzeCommandBase.cs

```diff
@@ -666,18 +666,30 @@ private async Task<bool> EnumerateFilesFromArtifactsProvider(TContext globalCont
                     continue;
                 }
 
-                if (artifact.SizeInBytes == 0)
+                long artifactSize = 0;
+                try
+                {
+                    artifactSize = artifact.SizeInBytes.Value;
+                }
+                catch (Exception e) when (e is IOException || e is ArgumentException)
+                {
+                    DriverEventSource.Log.ArtifactNotScanned(filePath, DriverEventNames.FilePathNotAllowed, 00, data2: null);
+                    Notes.LogFileSkipped(globalContext, filePath, e.Message);
+                    continue;
+                }
+
+                if (artifactSize == 0)
                 {
                     DriverEventSource.Log.ArtifactNotScanned(filePath, DriverEventNames.EmptyFile, 00, data2: null);
                     Notes.LogEmptyFileSkipped(globalContext, filePath);
                     continue;
                 }
 
-                if (!IsTargetWithinFileSizeLimit(artifact.SizeInBytes.Value, globalContext.MaxFileSizeInKilobytes))
```

Comment:
There is no functionality change in this paragraph. It just re-uses the variables defined before those statements.